receiver/mongodbreceiver/metrics.go

```diff
@@ -277,15 +300,15 @@ func (s *mongodbScraper) recordCacheOperations(now pcommon.Timestamp, doc bson.M
 	cacheMissesValue, err := parseInt(cacheMisses)
 	if err != nil {
 		errs.AddPartial(1, err)
-	} else {
-		s.mb.RecordMongodbCacheOperationsDataPoint(now, cacheMissesValue, metadata.AttributeTypeMiss)
+		return
 	}
+	s.mb.RecordMongodbCacheOperationsDataPoint(now, cacheMissesValue, metadata.AttributeTypeMiss)
 
 	tcr, err := dig(doc, []string{"wiredTiger", "cache", "pages requested from the cache"})
 	if err != nil {
 		errs.AddPartial(1, err)
 		s.logger.Debug("failed to parse total cache requests unable to calculate cache hits", zap.Error(err))
-		canCalculateCacheHits = false
```

Comment:
This is good cleanup work, but unrelated to the version checking. Please open another PR with this and other formatting / cleanup work.

Reply:
removed simply cache operation and formatting/cleanup work